pkg/engine/combat/damage.go

```diff
@@ -0,0 +1,58 @@
+package combat
+
+import (
+	"github.com/simimpact/srsim/pkg/engine/info"
+	"github.com/simimpact/srsim/pkg/engine/prop"
+)
+
+func (mgr *Manager) totalDamage(h *info.Hit, base float64, dmg float64, crit bool) float64 {
+	def := h.Defender.DEF()
+	def_mult := 1 - (def / (def + 200 + 10*float64(h.Attacker.Level())))
+
+	res := mgr.res(h)
+	vul := mgr.vul(h)
+
+	// toughness multiplier
+	toughness_multiplier := 0.9
+	if h.Defender.Stance() == 0 {
+		toughness_multiplier = 1
+	}
+
+	fatigue := 1 - h.Attacker.GetProperty(prop.Fatigue)
+	AllDamageReduce := 1 - h.Defender.GetProperty(prop.AllDamageReduce)
+	if AllDamageReduce < 0.01 {
+		AllDamageReduce = 0.01
+	}
+
+	crit_dmg := 1.0
+	if crit {
+		crit_dmg += h.Attacker.CritDamage()
+	}
+
+	total := base * dmg * def_mult * res * vul * toughness_multiplier * fatigue * AllDamageReduce * crit_dmg
+	return total
+}
+
+func (mgr *Manager) res(h *info.Hit) float64 {
+	// We don't currently have basic/ult dmg pen/res, dot pen/res, etc. If we do, we need to add it in here.
```

Comment:
get rid of comment, basic/skill/ult etc stuff is handled by implementation and not by separate stat (the `OnBeforeHit` that modify the properties prior to all this logic)